lib/twitch/client.rb

```diff
@@ -51,6 +51,7 @@ class Client
     # Used for refreshing User Access Token.
     def initialize(options = {})
       client_id = options[:client_id]
+      client_secret = options[:client_secret]
 
       @oauth2_client = TwitchOAuth2::Client.new(
         client_id: client_id, **options.slice(:client_secret, :redirect_uri, :scopes)
```

Comment:
Did you saw this line? `client_id` is assigned to local variable just because of re-using it below.